tests/src/test/java/alluxio/client/fs/LocalCacheFileInStreamIntegrationTest.java

```diff
@@ -0,0 +1,206 @@
+          .setProperty(PropertyKey.USER_BLOCK_SIZE_BYTES_DEFAULT, Constants.MB)
+          .setProperty(PropertyKey.USER_LOCAL_CACHE_ENABLED, true)
+          .build();
+
+  @Rule
+  public final ExpectedException mThrown = ExpectedException.none();
+
+  private FileSystemContext mFsContext;
+  private CacheManager mCacheManager;
+  private FileSystem mFileSystem;
+  private String mFilePath;
+
+  @Before
+  public void before() throws Exception {
+    mFsContext = FileSystemContext.create(mClusterResource.get().getClient().getConf());
+    mCacheManager = mFsContext.getCacheManager();
+    mFileSystem = mClusterResource.get().getClient(mFsContext);
+    mFilePath = PathUtils.uniqPath();
+  }
+
+  @After
+  public void after() throws Exception {
+    mFsContext.close();
+  }
+
+  @Test
+  public void read() throws Exception {
+    AlluxioURI path = new AlluxioURI(mFilePath);
+    FileSystemTestUtils.createByteFile(
+        mFileSystem, mFilePath, WritePType.MUST_CACHE, PAGE_SIZE_BYTES);
+    // read a file to populate the cache
+    try (FileInStream stream = mFileSystem.openFile(path)) {
+      assertTrue(BufferUtils.equalIncreasingByteArray(
+          PAGE_SIZE_BYTES, ByteStreams.toByteArray(stream)));
+    }
+    // verify locally cached data
+    URIStatus status = mFileSystem.getStatus(path);
+    try (ReadableByteChannel channel = mCacheManager.get(new PageId(status.getFileId(), 0))) {
+      assertNotNull(channel);
+      assertTrue(BufferUtils.equalIncreasingByteArray(
+          PAGE_SIZE_BYTES, ByteStreams.toByteArray(Channels.newInputStream(channel))));
+    }
+    mClusterResource.get().stopWorkers();
+    // verify reading from local cache
+    try (InputStream stream = mFileSystem.openFile(path)) {
+      assertTrue(BufferUtils.equalIncreasingByteArray(
+          PAGE_SIZE_BYTES, ByteStreams.toByteArray(stream)));
+    }
+  }
+
+  @Test
+  public void positionedRead() throws Exception {
+    AlluxioURI path = new AlluxioURI(mFilePath);
+    FileSystemTestUtils.createByteFile(
+        mFileSystem, mFilePath, WritePType.MUST_CACHE, PAGE_SIZE_BYTES);
+    try (FileInStream stream = mFileSystem.openFile(path)) {
+      byte[] buffer = new byte[PAGE_SIZE_BYTES / 4];
+      int bytesRead = stream.positionedRead(PAGE_SIZE_BYTES / 10, buffer, 0, buffer.length);
+      assertEquals(buffer.length, bytesRead);
+      assertTrue(BufferUtils.equalIncreasingByteArray(PAGE_SIZE_BYTES / 10, buffer.length, buffer));
+    }
+    // verify locally cached data
+    URIStatus status = mFileSystem.getStatus(path);
+    try (ReadableByteChannel channel = mCacheManager.get(new PageId(status.getFileId(), 0))) {
+      assertNotNull(channel);
+      assertTrue(BufferUtils.equalIncreasingByteArray(
+          PAGE_SIZE_BYTES, ByteStreams.toByteArray(Channels.newInputStream(channel))));
+    }
+    mClusterResource.get().stopWorkers();
+    // verify reading whole page from local cache
+    try (InputStream stream = mFileSystem.openFile(path)) {
```

Comment:
Once re removed `try (ReadableByteChannel channel = mCacheManager.get(new PageId(status.getFileId(), 0))) {...}` we should be asserting only on the pages just read?